cmd/nomos/vet/vet.go

```diff
@@ -43,6 +45,13 @@ func init() {
 	Cmd.Flags().BoolVar(&keepOutput, "keep-output", false,
 		`If enabled, keep the hydrated output`)
 
+	Cmd.Flags().IntVar(&threshold, "threshold", system.DefaultMaxObjectCount,
+		`If greater than zero, error if any repository contains more than the specified number of objects `)
+	// Set value to zero when `--threshold` is passed without a value.
+	// This requires the flag and value to be sent in the same argument, like
+	// `--threshold=1000`, instead of `--threshold 1000`.
+	Cmd.Flags().Lookup("threshold").NoOptDefVal = "0"
```

Comment:
This is different from the PR description 
> If you pass the option without a value, the default value of 1000 is used.

Reply:
The PR description is correct.

The behavior Mike suggested is that the flag behaves differently whether it's specified or not:
- When --threshold is specified without a value, the default is 1000 (configured by `DefaultMaxObjectCount` passed to `IntVar`)
- When --threshold is NOT specified, the default is 0 (configured by `NoOptDefVal`)

Reply:
> Set value to zero when `--threshold` is passed without a value.

Ah, the inline comment here is wrong.

Reply:
Apparently I got it backwards. `NoOptDefVal` specifies the default to use when the flag is specified with no value. Then the normal default is used when the flag is not specified at all.

Added tests to validate this, and fixed the comments. But the PR description is still correct.